api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Utils.java

```diff
@@ -30,4 +37,18 @@ static String randomAlphaNumString(int length) {
         .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
         .toString();
   }
+
+  public static JsonNode fetchDiscoveryData(URI discoveryEndpoint, HttpClient httpClient) {
+    URI uri = new UriBuilder(discoveryEndpoint).path(".well-known/openid-configuration").build();
+    HttpResponse response = httpClient.newRequest().path(uri.getPath()).get();
+    Status status = response.getStatus();
+    if (status != Status.OK) {
+      throw new HttpClientException("Discovery endpoint returned status code " + status.getCode());
+    }
+    JsonNode data = response.readEntity(JsonNode.class);
+    if (!data.has("issuer") || !data.has("authorization_endpoint")) {
```

Comment:
It looks like we expect the payload to follow the [OpenID Connect Discovery spec](https://openid.net/specs/openid-connect-discovery-1_0.html)... shall we make it explicit in docs?